syntax = "proto3";

import "packages/core/proto/timestamp.proto";

package planning;

// Types of trajectories generated by any of the planners. See:
// https://zippy.quip.com/r7s7ACWu6fkf
enum TrajectoryType {
    SQUARE = 0;
    TRAPEZOIDAL = 1;
    TRIANGULAR = 2;
    S = 3;
}

// A single trajectory element, comprising the system time, relative time,
// linear velocity component and the curvature. See:
// https://zippy.quip.com/61ZpA3tXRf2v.
message TrajectoryElement {
    core.Duration relative_time = 1;
    core.SystemTimestamp absolute_time = 2;
    float linear_velocity = 3;
    float curvature = 4;
    float arclength = 5;
}

// A full trajectory, comprising multiple TrajectoryElement and the associated
// type. The _counter_ variable identifies each serialized trajectory
// individually.
message Trajectory {
    repeated TrajectoryElement elements = 1;
    TrajectoryType type = 2;
    int64 counter = 3;
}
